src/packages/core/tests/server.test.ts

```diff
@@ -1257,58 +1261,287 @@ describe("server", () => {
+          ws.binaryType = "fragments";
+          const response: any = await new Promise(resolve => {
+            ws.on("open", () => {
+              ws.send(Buffer.from(JSON.stringify(jsonRpcJson)), {
+                binary: true
+              });
+            });
+            ws.on("fragment", data => {
+              console.log(data);
+            });
+            ws.on("message", resolve);
+          });
+
+          assert.strictEqual(Array.isArray(response), true);
+          const { result } = JSON.parse(Buffer.concat(response));
+          assert.strictEqual(result.structLogs.length, ops.length);
+        } finally {
+          Connector.BUFFERIFY_THRESHOLD = originalThreshold;
+          await teardown();
+        }
+      });
+
+      it("handles small+small+small fragment bufferification edge-case", () => {
+        // in this test we create many chunks, but they all fit in a single
+        // fragment, so we expect for bufferification to NOT kick in; it should
+        // be send with a regular `send` instead of `sendFragment`.
+        const chunks = [
+          // fits in a first fragment
+          Buffer.from("hello", "utf8"),
+          // fits in a first fragment
+          Buffer.from("world", "utf8"),
+          // fits in a first fragment
+          Buffer.from("!!!", "utf8")
         ];
-        // a silly "contract" we can trace later: PUSH 0, PUSH, 0, RETURN
-        const data = "0x" + ops.map(op => op.code + op.data).join("");
-        const hash = s.provider.send("eth_sendTransaction", [{ from, data }]);
-        await s.provider.once("message");
+        const expectedResponse = Buffer.concat(chunks);
+
+        let sendCallCount = 0;
+        const send = (
+          message: Buffer,
+          _isBinary?: boolean,
+          _compress?: boolean
+        ) => {
+          sendCallCount++;
+          // the received message should not be chunked further or joined with
+          // anything else
+          assert.strictEqual(
+            message.toString("utf8"),
+            expectedResponse.toString("utf8")
+          );
+          return true;
+        };
+        const mockWebsocket: any = {
+          send,
+          cork: (callback: () => void) => callback(),
+          // if we are sending a single chunk `send` must be used instead of
+          // sendFirstFragment/sendFragment/sendLastFragment
+          sendFragment: () => assert.fail("should not have used sendFragment"),
+          sendFirstFragment: () =>
+            assert.fail("should not have used sendFirstFragment"),
+          sendLastFragment: () =>
+            assert.fail("should not have used sendLastFragment")
+        };
+        const dataGenerator = function* () {
+          for (const chunk of chunks) yield chunk;
+        };
+        sendFragmented(mockWebsocket, dataGenerator(), false, chunkSize);
+        assert.strictEqual(sendCallCount, 1, "send called too many times!");
```

Comment:
```suggestion
        assert.strictEqual(sendCallCount, 1, "send called incorrect number of times!");
```

Or not enough times?